libcob/common.c

```diff
@@ -7812,9 +7814,10 @@ cob_expand_env_string (const char *strval)
 				}
 			}
 			if (penv != NULL) {
-				if ((strlen (penv) + j) > (envlen - 128)) {
-					env = cob_realloc (env, envlen, strlen (penv) + 256);
-					envlen = strlen (penv) + 256;
+				copylen = strlen (penv);
+				if (copylen + j > envlen - 128) {
+					env = cob_realloc (env, envlen, j + copylen + 256);
+					envlen = j + copylen + 256;
```

Comment:
please leave `envlen` as is, make  `copylen` a local variable here and have the new code with the copylen adjustment part of #112

Reply:
#112 ?